store/pruning/manager.go

```diff
@@ -74,22 +88,41 @@ func (m *Manager) HandleSnapshotHeight(height int64) {
 		return
 	}
 
+	m.logger.Debug("HandleSnapshotHeight", "height", height)
+
 	m.pruneSnapshotHeightsMx.Lock()
 	defer m.pruneSnapshotHeightsMx.Unlock()
 
-	m.logger.Debug("HandleSnapshotHeight", "height", height)
+	// remove from the in-flight list
+	if pos := slices.Index(m.inflightSnapshotHeights, height); pos != -1 {
+		m.inflightSnapshotHeights = append(m.inflightSnapshotHeights[:pos], m.inflightSnapshotHeights[pos+1:]...)
+	}
+
+	if m.initFromStore {
+		// drop the legacy state
+		m.pruneSnapshotHeights = m.pruneSnapshotHeights[1:]
+		m.initFromStore = false
+	}
+
 	m.pruneSnapshotHeights = append(m.pruneSnapshotHeights, height)
 	sort.Slice(m.pruneSnapshotHeights, func(i, j int) bool { return m.pruneSnapshotHeights[i] < m.pruneSnapshotHeights[j] })
+
 	k := 1
 	for ; k < len(m.pruneSnapshotHeights); k++ {
+		// anyone missing in the sequence means they are still in flight
 		if m.pruneSnapshotHeights[k] != m.pruneSnapshotHeights[k-1]+int64(m.snapshotInterval) {
 			break
 		}
 	}
+	// compact the height list when in-flight snapshots have landed, keep last height
 	m.pruneSnapshotHeights = m.pruneSnapshotHeights[k-1:]
 
-	// flush the updates to disk so that they are not lost if crash happens.
-	if err := m.db.SetSync(pruneSnapshotHeightsKey, int64SliceToBytes(m.pruneSnapshotHeights)); err != nil {
+	if n := len(m.pruneSnapshotHeights); n > 100 {
```

Comment:
why was this value chosen?

Reply:
It is an arbitrary value. I have removed this block as the warning is not really actionable. People could restart the node but it probably creates more confusion than it helps.